tests/unity/unity_test.cmake

```diff
@@ -0,0 +1,135 @@
+#
+# Copyright (c) 2019 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+include($ENV{ZEPHYR_BASE}/cmake/app/boilerplate.cmake NO_POLICY_SCOPE)
+
+set(UNITY_PRODUCTS_DIR "${APPLICATION_BINARY_DIR}/unity")
+set(CMOCK_PRODUCTS_DIR "${UNITY_PRODUCTS_DIR}/mocks")
+set(CMOCK_DIR "${ZEPHYR_BASE}/../cmock")
+set(RUBY_EXEC "/usr/bin/ruby")
+file(MAKE_DIRECTORY "${UNITY_PRODUCTS_DIR}")
+
+# Generate cmock for provided header file.
+function(cmock_generate header_path dst_path)
+  file(MAKE_DIRECTORY "${dst_path}")
+
+  execute_process(
+    COMMAND ${RUBY_EXEC}
+    ${CMOCK_DIR}/lib/cmock.rb
+    --mock_prefix=mock_
+    --plugins=return_thru_ptr\;ignore_arg\;ignore\;callback\;array
+    --mock_path=${dst_path}
+    ${header_path}
+    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+    RESULT_VARIABLE op_result
+    OUTPUT_VARIABLE output_result
+  )
+
+  target_include_directories(app PRIVATE ${CMOCK_PRODUCTS_DIR})
+  #target_include_directories(app PRIVATE ${dst_path})
+  file(GLOB app_sources ${dst_path}/*.c)
+  target_sources(app PRIVATE ${app_sources})
+endfunction()
+
+# Generate test runner file.
+function(test_runner_generate test_file_path)
+    get_filename_component(test_file_name "${test_file_path}" NAME)
+    set(output_file "${UNITY_PRODUCTS_DIR}/runner/runner_${test_file_name}")
+    file(MAKE_DIRECTORY "${UNITY_PRODUCTS_DIR}/runner")
+
+    execute_process(
+      COMMAND ${RUBY_EXEC}
+      ${CMOCK_DIR}/vendor/unity/auto/generate_test_runner.rb
+      ${test_file_path} ${output_file}
+      WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+      RESULT_VARIABLE op_result
+    )
+    file(GLOB app_sources ${UNITY_PRODUCTS_DIR}/runner/*.c)
+    target_sources(app PRIVATE ${app_sources})
+
+    message(STATUS "Generating test runner ${unity_runner_arg3}")
+endfunction()
+
+# Add --wrap linker option for each function listed in the input file.
+function(cmock_linker_trick func_name_path)
+    file(READ "${func_name_path}" contents)
+    string(REGEX REPLACE "\n" ";" contents "${contents}")
```

Comment:
Use file(STRINGS, instead of file and string.

https://cmake.org/cmake/help/v3.14/command/file.html#strings

Reply:
i cannot find how if it's possible to do file(STRINGS REGEX REPLACE..).